bofire/strategies/doe_strategy.py

```diff
@@ -20,22 +32,138 @@ def __init__(
     ):
         super().__init__(data_model=data_model, **kwargs)
         self.formula = data_model.formula
+        self.data_model = data_model
+        self._partially_fixed_experiments_for_next_design = None
+
+    def tell(
+        self,
+        experiments: pd.DataFrame,
+        replace: bool = False,
+    ) -> None:
+        """This function passes new experimental data to the optimizer
+
+        Args:
+            experiments (pd.DataFrame): DataFrame with experimental data
+            replace (bool, optional): Boolean to decide if the experimental data should replace the former DataFrame or if the new experiments should be attached. Defaults to False.
+        """
+
+        self._partially_fixed_experiments_for_next_design = experiments[
+            experiments.isnull().any(axis=1)
+        ][self.domain.get_feature_keys(includes=Input)]
+        experiments = experiments[experiments.notnull().all(axis=1)]
+
+        if len(experiments) == 0:
+            return
+        if replace:
+            self.set_experiments(experiments=experiments)
+        else:
+            self.add_experiments(experiments=experiments)
+        # we check here that the experiments do not have completely fixed columns
+        cleaned_experiments = (
+            self.domain.outputs.preprocess_experiments_all_valid_outputs(
+                experiments=experiments
+            )
+        )
+        for feature in self.domain.inputs.get_fixed():
+            if (cleaned_experiments[feature.key] == feature.fixed_value()[0]).all():  # type: ignore
+                raise ValueError(
+                    f"No variance in experiments for fixed feature {feature.key}"
+                )
+        self._tell()
+
+    def _tell(self) -> None:
+        self.set_candidates(
+            self.experiments[self.domain.get_feature_keys(includes=Input)]
+        )
+        raise NotImplementedError(
```

Comment:
Also this is a bit cumbersome, the idea is that one has a generic `tell` method in `Strategy`, which is then calling a strategy dependent `_tell`. In principal, I do not see why a tell is needed at all, we have already the pending candidates to take previous experiments into account ...

Reply:
Whoops. I told Ufuk to implement tell. My bad. @ufukguenes let us use set_candidates() then instead.